lib/com/vaadin/polymer/PolymerEvent.java

```diff
@@ -0,0 +1,39 @@
+package com.vaadin.polymer;
+
+import jsinterop.annotations.JsProperty;
+import jsinterop.annotations.JsType;
+import elemental2.dom.EventInit;
+import elemental2.dom.EventTarget;
+
+import static jsinterop.annotations.JsPackage.GLOBAL;
+
+
+/**
+ * PolymerEvent is a class over the Element implementation of elemental2 adding
+ * some properties present in Polymer Events, or missing in the elemental2 API.
+ */
+@JsType(isNative=true, namespace=GLOBAL, name="Event")
+public class PolymerEvent extends elemental2.dom.Event {
+
+    public PolymerEvent(String type, EventInit eventInitDict) {
+        super(type, eventInitDict);
+    }
+
+    public PolymerEvent(String type) {
+        super(type);
+    }
+
+    @JsProperty
+    public native String getType();
+
+    @JsProperty
+    public native Detail getDetail();
+
+    @JsProperty
+    public native EventTarget getTarget();
+
+    @JsType(isNative=true)
+    public interface Detail {
```

Comment:
A bit late, but I think here it is better to use `jsinterop.base.Any` as `getDetail()` result and not to create this interfaces...

Reply:
@ibaca  this is an example of why we are in snapshot in the gwt-polymer-elements. We still can improve the elemental2 migration. Feel free to send a PR or open an issue.

Reply:
yep, I have tried, maybe I finally end up fixing it but the current project-layout is not easy to modify bc 1. requires node+gulp+js knowledge 👎 (my bad) and even with that the project is not a java one so if I modify I need to generate and goes back to this project. Probably this is good enough, I'm just complaining 😉 to justify the next proposal. For the next version, polymer 2  using (now renamed) polymer-analyzer don't you thing might be easier to develop this tool saving only the JSON model to some file spec.json, and develop the whole generator in java+javapoet, so this same project can contain the common utils.

Ok, so... an as-small-as-possible JS tool to create all needed metadata in a spec.json and a second java project to, from this spec.json, generating the Java API.

Reply:
Yes that is the basic idea to create java server API from polymer in vaadin framework, we use analyzer for json and roaster for java generator.

I tried analyzer in gwt-api-generator a year ago but it had an issue not giving type info so I had to postpone see #68 . Since it was fixed some months ago, I would continue with my work when I find a while.

Not sure still about replacing the current js + template generation by a java solution, probably it makes sense to reuse some code from elemental2 instead.

BTW we used JS for many reasons: simplicity, less code, everything in one cli, any polymer developer or vendor can deliver it's java API without any knowledge of the java ecosystem, and developers in the vaadin elements team are more familiar with it ;-).